rush-plugins/rush-azure-storage-build-cache-plugin/src/AzureAuthenticationBase.ts

```diff
@@ -233,25 +247,36 @@ export abstract class AzureAuthenticationBase {
    */
   protected abstract _getCacheIdParts(): string[];
 
-  protected abstract _getCredentialFromDeviceCodeAsync(
+  protected abstract _getCredentialFromTokenAsync(
     terminal: ITerminal,
-    deviceCodeCredential: DeviceCodeCredential
+    tokenCredential: TokenCredential
   ): Promise<ICredentialResult>;
 
-  private async _getCredentialAsync(terminal: ITerminal): Promise<ICredentialResult> {
+  private async _getCredentialAsync(
+    terminal: ITerminal,
+    loginFlow: LoginFlowType
+  ): Promise<ICredentialResult> {
     const authorityHost: string | undefined = AzureAuthorityHosts[this._azureEnvironment];
     if (!authorityHost) {
       throw new Error(`Unexpected Azure environment: ${this._azureEnvironment}`);
     }
 
-    const deviceCodeCredential: DeviceCodeCredential = new DeviceCodeCredential({
-      ...this._additionalDeviceCodeCredentialOptions,
-      authorityHost: authorityHost,
-      userPromptCallback: (deviceCodeInfo: DeviceCodeInfo) => {
-        PrintUtilities.printMessageInBox(deviceCodeInfo.message, terminal);
-      }
-    });
+    let tokenCredential: TokenCredential;
+    if (loginFlow === 'InteractiveBrowser') {
+      tokenCredential = new InteractiveBrowserCredential({
+        ...this._additionalInteractiveCredentialOptions,
+        authorityHost: authorityHost
+      });
+    } else {
+      tokenCredential = new DeviceCodeCredential({
+        ...this._additionalDeviceCodeCredentialOptions,
+        authorityHost: authorityHost,
```

Comment:
```suggestion
    const interactiveCredentialOptions: (
      | InteractiveBrowserCredentialNodeOptions
      | InteractiveBrowserCredentialInBrowserOptions
    ) &
      DeviceCodeCredentialOptions = {
      ...this._additionalInteractiveCredentialOptions,
      authorityHost
    };
    
    if (loginFlow === 'InteractiveBrowser') {
      tokenCredential = new InteractiveBrowserCredential(interactiveCredentialOptions );
    } else {
      tokenCredential = new DeviceCodeCredential({
        ...interactiveCredentialOptions,
```
And yes, I know this conflicts with my other suggestion.